pallets/parachain-staking/src/delegation_requests.rs

```diff
@@ -248,6 +248,7 @@ impl<T: Config> Pallet<T> {
 				<DelegationScheduledRequests<T>>::insert(collator, scheduled_requests);
 				if leaving {
 					<DelegatorState<T>>::remove(&delegator);
+					let _ = T::ScoreUpdater::clear_score_for(&delegator);
```

Comment:
Can you also check what if the user unstakes all the amount (`bondLess`) instead of revoking the delegation? Will the hook be triggered too?

Reply:
let me check 👍🏼

Reply:
> If I search <DelegatorState<T>>::remove in the code, there's a fourth place in types.rs - can you check if we need to trigger the hook there too?

Yes, the delegator state is also removed when a bottom delegation is kicked out. I've updated the code to trigger the call.

> Can you also check what if the user unstakes all the amount (bondLess) instead of revoking the delegation? Will the hook be triggered too?

The are checks that prevent delegators and candidates to decrease the bond with an amount that is greater than the bonded amount, so in these cases we don't need to trigger it

Reply:
Does it mean the user can only unstake all money with the call to `revoke`?

Reply:
From the delegator's perspective it can be done with revoke and leave_delegators. The stake is also returned on `leave_candidates` and when the list of delegators is full and another user bonds more, the bottom delegation is kicked out